### PR TITLE
Add factory methods for creating schema-valid test data

### PR DESCRIPTION
**Pull request by eslavich**
This adds some methods for generating nodes that are valid against their schemas.  For example, we can create a `WfiImage` like this:

```python
from roman_datamodels.testing import factories

wfi_image = factories.create_wfi_image()
```

Attributes can be overridden by passing in keyword arguments:

```python
aperture = factories.create_aperture(position_angle=1.234)
meta = factories.create_meta(aperture=aperture)
wfi_image = factories.create_wfi_image(meta=meta)
```

Otherwise you just get random numbers.  I'm sure I didn't do a good job of selecting reasonable values, but people can refine that over time.

If this turns out to be a hassle to maintain we might look into generating data from the schema itself, but that sounds pretty hard to me.  I have enough tests in here that if someone adds or updates a schema we'll get prompted to update the factory methods.

### CODE REVIEW

**Comment by eslavich** [comment]
Tests should pass once https://github.com/spacetelescope/rad/pull/17 is merged

**Review by perrygreenfield — APPROVED** [review]
Still looks good to me. I think refinement of the range of random values should be left to future PRs, and will likely be continually  updated as more information is obtained. In some sense, I see this sort of information as augmenting what is in the schemas (though min/max cases could be put into schemas eventually).

File: src/roman_datamodels/testing/factories.py [hunk 1]
```diff
@@ -0,0 +1,854 @@
...
+    ----------
+    **kwargs
+        Additional or overridden attributes.
+
+    Returns
+    -------
+    roman_datamodels.stnode.VelocityAberration
+    """
+    raw = {
+        "ra_offset": _random_float(),
```
[the remark this answers is not in the document]
Comment (eslavich) [reply src/roman_datamodels/testing/factories.py:670]: I'll add a TODO comment here

File: src/roman_datamodels/testing/factories.py [hunk 2]
```diff
@@ -0,0 +1,854 @@
...
+        Additional or overridden attributes.
+
+    Returns
+    -------
+    roman_datamodels.stnode.Program
+    """
+    raw = {
+        "category": _random_string("Cat ", 6),
+        "continuation_id": _random_positive_int(),
+        "pi_name": _random_string("PI name ", 100),
```
[the remark this answers is not in the document]
Comment (eslavich) [reply src/roman_datamodels/testing/factories.py:610]: I did add `subcategory`, but `pi_name` is still present in the schema:

https://github.com/spacetelescope/rad/blob/main/src/rad/resources/schemas/program-1.0.0.yaml#L19

File: src/roman_datamodels/testing/factories.py [hunk 3]
```diff
@@ -0,0 +1,854 @@
...
+
+    Returns
+    -------
+    roman_datamodels.stnode.FlatRef
+    """
+    raw = {
+        "data": _random_array_float32(),
+        # TODO: Update this to _random_array_uint16() once the schema is fixed.
+        # See https://github.com/spacetelescope/rad/issues/8
+        "dq": _random_array_float32(),
```
[the remark this answers is not in the document]
Comment (eslavich) [reply src/roman_datamodels/testing/factories.py]: This has been updated to uint32

File: src/roman_datamodels/testing/factories.py [hunk 4]
```diff
@@ -0,0 +1,854 @@
...
+    -------
+    roman_datamodels.stnode.Exposure
+    """
+    raw = {
+        "count": _random_positive_int(),
+        "data_problem": _random_bool(),
+        "duration": _random_positive_float(),
+        "effective_exposure_time": _random_positive_float(),
+        "elapsed_exposure_time": _random_positive_float(),
+        "end_time": _random_utc_timestamp(),
```
[the remark this answers is not in the document]
Comment (eslavich) [reply src/roman_datamodels/testing/factories.py]: That's good news, we need the motivation

File: src/roman_datamodels/testing/factories.py [hunk 5]
```diff
@@ -0,0 +1,854 @@
...
+    required by the schema.
+
+    Parameters
+    ----------
+    **kwargs
+        Additional or overridden attributes.
+
+    Returns
+    -------
+    roman_datamodels.stnode.WfiScienceRaw
```
[the remark this answers is not in the document]
Comment (eslavich) [reply src/roman_datamodels/testing/factories.py:801]: I'll leave the TODO in for now, since the schema currently requires 4D.

**Pull request merged**